lib/matplotlib/colorbar.py

```diff
@@ -880,17 +880,20 @@ def __init__(self, ax, mappable, **kw):
         mappable.autoscale_None()
 
         self.mappable = mappable
+        kw = cbook.normalize_kwargs(kw, forbidden=('cmap', 'norm'))
         kw['cmap'] = cmap = mappable.cmap
-        kw['norm'] = norm = mappable.norm
+        kw['norm'] = mappable.norm
 
         if isinstance(mappable, contour.ContourSet):
             CS = mappable
+            override_list = ('alpha', 'boundaries', 'values', 'extend',
+                             'ticks', 'filled')
+            kw = cbook.normalize_kwargs(kw, forbidden=override_list)
             kw['alpha'] = mappable.get_alpha()
+            kw['ticks'] = ticker.FixedLocator(CS.levels, nbins=10)
```

Comment:
I think we need to preserve the "setdefault" behavior to allow flexibility in setting ticks.